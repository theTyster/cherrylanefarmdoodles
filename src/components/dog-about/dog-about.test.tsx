import { env } from "cloudflare:test";
import { describe, test, expect } from "vitest";
import DogAbout, { dogAboutQuery } from "./dog-about";
import React from "react";

describe("DogAbout Component", async () => {
  const data = await env.dogsDB
    .prepare(dogAboutQuery("mother", "father"))
    .bind("Max")
    .first<DogAboutData>();

  if (!data) throw new Error("No data found ðŸ˜¢");

  const { Group_Photos } = data;

  const dogData = {
    adultName: data.adultName,
    breeder: data.breeder,
    birthday: data.birthday,
    eyecolor: data.eyecolor,
    isRetired: data.isRetired,
    favActivities: data.favActivities,
    weight: data.weight,
    energyLevel: data.energyLevel,
    gender: data.gender as "M" | "F",
    noseColor: data.noseColor,
    coatColor: data.coatColor,
    personality: data.personality,
    Headshots_Lg: data.Headshots_Lg,
  } as DogAbout_DogData;

  const partnerData = {
    partnerName: data.partnerName,
    partnerBreeder: data.partnerBreeder,
    partnerBirthday: data.partnerBirthday,
    partnerPhoto: data.partnerPhoto,
  } as DogAbout_PartnerData;

  const render = (
    <DogAbout
      dogData={dogData}
      partnerData={partnerData}
      Group_Photos={Group_Photos}
    />
  );

  test("DogAbout component should be defined", () => {
    expect.soft(render).toBeTruthy();
  });
  test("data passed into DogAbout component should be consistent", () => {
    expect.soft(render).toMatchSnapshot();
  });
});
