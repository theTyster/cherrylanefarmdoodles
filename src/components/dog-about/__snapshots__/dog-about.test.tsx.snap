// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DogAbout Component > data passed into DogAbout component should be consistent 1`] = `
<DogAbout
  Group_Photos="https://media.cherrylanefarmdoodles.com/preview/jn5sLxWFkmx-hVd5t6q8ynBkxQUqw7lRnO8wh2qiMWsrLrEYLNejI2gYqzvABXeSU9sXZnhueCNs_lcCs1b2wg0__ZkCtIF8LqxozTwEadw00gyxXTHrLOMXR2Iu6H5BAPdjWSM994cQ-F91Qir0ZmP-nU8N8NY4mKJIbmBJWk8PY_mNPQ?iv=FHWo6K4yEp82pQse&k=GQD2siGEchj3X83Df5RyKJJfSxMgRd53vV08OAYvSBoE2Oc3RcFiTlGjvzVhBZG5x0oAKgqRigxsqjsIOUGJ-2bt_NWo-KuM8TvGOOUCdF0mpTSZDxO4obgvf4RH3zKoIO-aQJ7U9D9Jaiph-H2BjQ16iSAN3fjuW9gfx2UyyaQw3FbaR_R9Vg"
  dogData={
    {
      "Headshots_Lg": "https://media.cherrylanefarmdoodles.com/preview/zrNfo4XuOoQrvmPsvmrXq1D-z3thkqZiAfSeI-2MkJlkY_gKOgtHtDqtUkdwFUwmDRrp3Le_XhGVcGcXrLk2woeG8sHSTYCTEKhIp53z4m95v21CNY5dbI6Em9uMZrN_PNozy8qyYbBmOfmzAymP4J8k5rLwmt4vwfaLudqH_RhagkfC8VcalO7RJmNdVA?iv=Ir9podsTop8FkYAR&k=mfPsyZWGm2VDYJuiMh3xdFlsC4Wk8Ag__7UVQEE4CD3y4YNcTzoVbC8XtpzNV3JrspYFLaK_3YMPsuft2_6cNxV2F81xM30O8sKg_SRzlbQEPRAad90Xaq-7e-_o3h2nF0ArE1am6JXKRVWjp7ZlF-wQLxVy7SRmtJ2Xi0noKYcTTLOZCbPlaQ",
      "adultName": "Max",
      "birthday": "2018-05-12",
      "breeder": "Breeder A",
      "coatColor": "Black",
      "energyLevel": "High",
      "eyecolor": undefined,
      "favActivities": "Playing fetch and running.",
      "gender": "M",
      "isRetired": 0,
      "noseColor": "White",
      "personality": "Intelligent and loyal.",
      "weight": 25,
    }
  }
  partnerData={
    {
      "partnerBirthday": "2019-02-28",
      "partnerBreeder": "Breeder B",
      "partnerName": "Bella",
      "partnerPhoto": "https://media.cherrylanefarmdoodles.com/preview/JdMOnq6Bu9CAE6fIkse7FFQHsOFll4f8fpfqyvVXg6ICKf0Dr6KElDJzgOPWJ3KbNyM8caRz9i6Bz-6L25leQT893ddKxw5vE_SQI4J4-j7ZZZRbgk9HdXdXMpZHV6OPddIQN_tVJNNmN-Uerw332ntDnh_e_q5kt5cU6-sc7nmzG8myMGtLFmHgrnXewig5?iv=qjsHSx3UZz5dn2QX&k=PoCgOrrPODVfwrce-nlF2aqsEK8TLevYbbbR5lddT1UX-PUynRz0_Q1ERdbDR8jOzDyBGIq6zPJo8ObpXns_hiFyR7FXKW2slozuKp7KXlnho2SnFL__YBluinBWdLDGRGn8qy7X8dnv2iXkoWoyE69CGIc4-aJFejNSlJSD7SC9wHtg_sb4tg",
    }
  }
/>
`;
